packages/core/src/validate/run.ts

```diff
@@ -241,13 +243,35 @@ function* getConstructorErrors(contractDef: ContractDefinition, decodeSrc: SrcDe
   }
 }
 
+function initCaches() {
+  const delegateCallCache: Cache = {
+    mainContractErrors: new Map<number, ValidationErrorOpcode[]>(),
+    inheritedContractErrors: new Map<number, ValidationErrorOpcode[]>(),
+    visitedNodeIds: new Map<number, Scope>(),
+  };
+  const selfDestructCache: Cache = {
+    mainContractErrors: new Map<number, ValidationErrorOpcode[]>(),
+    inheritedContractErrors: new Map<number, ValidationErrorOpcode[]>(),
+    visitedNodeIds: new Map<number, Scope>(),
+  };
+  return { delegateCallCache, selfDestructCache };
+}
+
+interface Cache {
```

Comment:
I would consider renaming `Cache` to `OpcodeCache` to be a little more specific. Also `initCaches` to `initOpcodeCaches`, etc.